src/main/java/io/stargate/sgv2/jsonapi/service/shredding/collections/DocumentShredder.java

```diff
@@ -313,15 +315,29 @@ private void traverseVector(JsonPath path, JsonNode value, DocumentShredderListe
     if (value.isNull()) {
       return;
     }
-    if (!value.isArray()) {
+    // should be either array or object
+    if (!value.isArray() && !value.isObject()) {
       throw ErrorCodeV1.SHRED_BAD_DOCUMENT_VECTOR_TYPE.toApiException(
           value.getNodeType().toString());
     }
-    ArrayNode arr = (ArrayNode) value;
-    if (arr.size() == 0) {
-      throw ErrorCodeV1.SHRED_BAD_VECTOR_SIZE.toApiException();
+    // e.g. "$vector": [0.25, 0.25]
+    if (value.isArray()) {
+      ArrayNode arr = (ArrayNode) value;
+      if (arr.size() == 0) {
+        throw ErrorCodeV1.SHRED_BAD_VECTOR_SIZE.toApiException();
+      }
+      callback.shredVector(path, arr);
+    }
+    // e.g. "$vector": {"$binary": "c3VyZS4="}
+    if (value.isObject()) {
+      ObjectNode obj = (ObjectNode) value;
+      final Map.Entry<String, JsonNode> entry = obj.fields().next();
+      JsonExtensionType keyType = JsonExtensionType.fromEncodedName(entry.getKey());
+      if (keyType != BINARY) {
+        throw ErrorCodeV1.SHRED_BAD_DOCUMENT_VECTOR_TYPE.toApiException(entry.getKey());
+      }
+      callback.shredVector(path, obj);
```

Comment:
as per comment below: instead of passing `obj`, pass either `JsonNode` for key `$binary`; or, maybe better text, textual value of that node (which is just `String`).